dipy/reconst/ivim.py

```diff
@@ -153,20 +166,18 @@ def nlls_fit(data, gtab, guess_params):
 
     Returns
     -------
-    nlls_params: the S0, f, D_star, D value for each voxel.
+    ivim_params: the S0, f, D_star, D value for each voxel.
 
     """
     flat_data = data.reshape((-1, data.shape[-1]))
     # Flatten for the iteration over voxels:
     bvals = gtab.bvals
     ivim_params = np.empty((flat_data.shape[0], 4))
     for vox in range(flat_data.shape[0]):
-        popt, pcov = curve_fit(ivim_function, bvals, flat_data[vox],
-                               guess_params[vox])
-        ivim_params[vox, :4] = popt
+        res = leastsq(_ivim_error,
```

Comment:
On Thu, Jun 2, 2016 at 9:16 AM, Eric Peterson <notifications@github.com>
wrote:

> In dipy/reconst/ivim.py
> <https://github.com/nipy/dipy/pull/1058#discussion_r65571187>:
>
> >
> >      """
> >      flat_data = data.reshape((-1, data.shape[-1]))
> >      # Flatten for the iteration over voxels:
> >      bvals = gtab.bvals
> >      ivim_params = np.empty((flat_data.shape[0], 4))
> >      for vox in range(flat_data.shape[0]):
> > -        popt, pcov = curve_fit(ivim_function, bvals, flat_data[vox],
> > -                               guess_params[vox])
> > -        ivim_params[vox, :4] = popt
> > +        res = leastsq(_ivim_error,
>
> Yeah, I think that's a red herring though. I bet if you print or plot the
> results you'd find that it's changing the b-values to fit the curve and not
> the curve to fit the b-values.
>
> But the function to be optimized is `_ivim_error`, which does have the
model parameters as its first input. Or I might be missing what you mean
here.


> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/nipy/dipy/pull/1058/files/c955cab175ec32c335e7ccd641ef9c46973418fa..834c2da60b66d8daf228cd45cdddf60152af9079#r65571187>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAHPNrD-MANHK2LmRY5baHGnEnL_dKiLks5qHwHrgaJpZM4Imo2R>
> .
>